azure-pipelines-deploy-template.yml

```diff
@@ -81,16 +78,83 @@ jobs:
       runOnce:
         deploy:
           steps:
-          - task: AzureResourceGroupDeployment@2
-            displayName: 'Azure Deployment:Create Or Update Resource Group action on $(resourceGroupName)'
+          - task: AzurePowerShell@3
+            displayName: 'Azure PowerShell Script - Set Deployment Type for Resource Group $(resourceGroupName)'
             condition: succeeded()
+            env:
+              SYSTEM_ACCESSTOKEN: $(system.accesstoken)
+              COMMIT_HASH: $(build.sourceVersion)
+              CURRENT_BRANCH: $(build.sourceBranchName)
+            inputs:
+              azureSubscription: '${{parameters.subscriptionName}}'
+              ScriptType: InlineScript
+              azurePowerShellVersion: LatestVersion
+              Inline: |
+                Param(
+                  [string] $resourceGroupName = "$(resourceGroupName)",
+                  [string] $environment = "${{parameters.environment}}"
+                )
+
+                $fullDeployment = $false
+                $deploymentFiles = @("azure-pipelines.yml", "azure-pipelines-release.yml", "azure-pipelines-deploy-template.yml", "azure/template.json", "azure/containers.json")
+                $encodedAccessToken = [System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes($env:SYSTEM_ACCESSTOKEN))
+                $variableGroupLastEdit = @()
+
+                #Get the timestamp of the last deployment to the defined Azure environment from Azure API
+                $lastDeployment = (Get-AzureRmResourceGroupDeployment -ResourceGroupName $resourceGroupName | Where-Object {$_.DeploymentName -Like "template-*"})[0].Timestamp
+
+                #Get the commit hash for the current version deployed in the environment
+                $currentDeployedCommit = (Get-AzureRmResourceGroup -Name $resourceGroupName).Tags.Version
+                 
+                #Get the timestamp of the last update to the variable groups corresponding to the environment defined
+                $variableGroupUri = "https://dev.azure.com/dfe-ssp/Become-A-Teacher/_apis/distributedtask/variablegroups"
+                $variableGroups = Invoke-RestMethod -Method Get -Uri $variableGroupUri -Headers @{Authorization = "Bearer $env:SYSTEM_ACCESSTOKEN"}
+                $variableGroupLastEdit += ($variableGroups.value | Where-Object {$_.name -like "APPLY - ENV - $environment"}).modifiedOn
+                $variableGroupLastEdit += ($variableGroups.value | Where-Object {$_.name -eq "APPLY - Shared Variables"}).modifiedOn
+                $variableGroupLastEdit += ($variableGroups.value | Where-Object {$_.name -eq "Docker Shared Variables"}).modifiedOn
```

Comment:
@duncanjbrown I'm not sure how clever we can necessarily make that because if we assume the use of "APPLY" groups we're hoping that anyone adding a new group remembers to use that naming convention and I won't necessarily know if any new group is specific to all or a subset of the environments.
Maybe it might be better to add a section on variable groups in the README to cover creation of new variable groups and the fact that this script may need to be updated if changes are made.
I don't expect creation of new variable groups to be a particularly common thing.

Reply:
@tvararu given @duncanjbrown and @tijmenb are on holiday now, are you happy with my proposal and the documentation I've added for us to run through merging this on Monday?

Reply:
Yes